### DISCUSSION TITLE
Name of the signer not showing some pdf

### DISCUSSION THREAD

**Discussion by epdxb** [MatthiasValvekens/pyHanko#120] [General]
Hi Matthias,

I hit an roadblock as i observed on some pdf files name of signer is not showing as specified in 'name' value of PdfSignatureMetadata object, especially on the PDF that i added some text or images using reportlab.pdfgen. 

This the sample code to replicate the issue. 

When signing the Input1.pdf don't have any issue, when i open the singed file on adobe viewer is showing signer. Please check below screenshot of final signed file

![Photo_Editor_Pixlr_X_free_image_editing_online(1)](https://user-images.githubusercontent.com/74245498/173295816-d922f6b2-527a-4656-a198-4ccf97f85226.png)

But when signing Input2.pdf having the issue, when i open the singed file on adobe viewer is hot showing signer. Please check below screenshot of final signed file

![Photo_Editor_Pixlr_X_free_image_editing_online(2)](https://user-images.githubusercontent.com/74245498/173296931-23caafb5-2961-42ec-9d41-64ff2dbfd312.png)

**Comment by MatthiasValvekens** [comment]
That's very strange. I could imagine differences arising if you'd be using different certs or something, but this is indeed strange. Can you share output files? I would like to check if the **Name** entry is actually being written.

**Comment by epdxb** [comment]
Hi Matthias,

Thanks for the reply. These are final signed documents

[Signed_Input1.pdf](https://github.com/MatthiasValvekens/pyHanko/files/8895817/Signed_Input1.pdf)

[Signed_Input2.pdf](https://github.com/MatthiasValvekens/pyHanko/files/8895821/Signed_Input2.pdf)

**Reply by MatthiasValvekens** [reply]
Hi @epdxb, thanks for sharing those documents. I'm afraid I don't know why Acrobat is behaving the way it is. The `/Name` entry is (correctly) set to the same value in both cases, so there must be _something_ in the structure of either document that is causing Acrobat to process things differently. Could you try re-saving both files with Acrobat/Adobe Reader and investigate whether you can still reproduce the difference?

Either way, I'm fairly sure that this is an Acrobat quirk.


Unrelated observation: judging from your sample code, I find it somewhat curious that you seem to have gotten your hands on an AATL-backed document signing certificate for which the private key resides in a file on disk (i.e. not in an HSM)... That's against AATL policy (see clause EE4 in the [technical requirements](https://helpx.adobe.com/content/dam/help/en/acrobat/kb/approved-trust-list2/jcr_content/main-pars/download_section/download-1/aatl_requirements20.pdf)). Either your CA (DigiCert) is playing fast and loose with the rules here, or you might be (unwittingly) violating the terms of your CA's subscriber agreement. That's not my problem, but you might want to carefully review the terms before using this key pair for anything serious. You might even need to have the cert reissued for a new (properly managed) key.

**Comment by epdxb** [comment]
Hi Matthias, 
Thanks for the support.  We check on the different PDF files and other than reportlab.pdfgen we will try different python library to write in PDF file.
